src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -112,30 +122,67 @@ def on_time_step(self, event: Event):
         choose which households move;
         move those households to a new address
         """
-        households = self.population_view.subview(
-            ["household_id", "relation_to_household_head", "address", "zipcode"]
-        ).get(event.index)
+        households = self.population_view.get(event.index)
         household_heads = households.loc[
             households["relation_to_household_head"] == "Reference person"
         ]
         households_that_move = self.addresses.determine_if_moving(
             household_heads["household_id"], self.household_move_rate
         )
 
-        if len(households_that_move) > 0:
+        # Find households that move abroad and separate subsets of state table
+        households_that_move_aboard_idx = self.determine_households_moving_abroad(
+            households_that_move,
+            self.proportion_households_leaving_country,
+            event
+        )
+        moving_abroad_households = households_that_move.loc[households_that_move_aboard_idx]
+        moving_domestic_households = households_that_move.loc[
+            ~households_that_move.index.isin(moving_abroad_households.index)
+        ]
+        abroad_moving_households = households.loc[
+            households["household_id"].isin(moving_abroad_households)
+        ]
+        domestic_moving_households = households.loc[
+            households["household_id"].isin(moving_domestic_households)
+        ]
+
+        # Process households moving abroad
+        if len(abroad_moving_households) > 0:
+            abroad_moving_households["exit_time"] = event.time
+            abroad_moving_households["tracked"] = False
+
+        if len(domestic_moving_households) > 0:
             address_map, zipcode_map = self.addresses.get_new_addresses_and_zipcodes(
-                households_that_move, state=metadata.US_STATE_ABBRV_MAP[self.location].lower()
+                moving_domestic_households, state=metadata.US_STATE_ABBRV_MAP[self.location].lower()
```

Comment:
do you need this to be wrapped in an `if`? If not, I'd recommend removing it (but I admittedly don't know how `addresses.get_new_addresses_and_zipcodes` behaves if domestic_moving_households is an empty dataframe